### PR TITLE
Add test engine filter to documentation

### PR DESCRIPTION
**Pull request by blindpirate**
As suggested by @eriwen in #4523 , this PR adds a section to tell users how to use `includeEngine` and `excludeEngine` for JUnit 5.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/javaPlugin.adoc [hunk 1]
```diff
@@ -755,12 +755,6 @@ The following code enables JUnit Platform support in `build.gradle`:
 
 See api:org.gradle.api.tasks.testing.Test#useJUnitPlatform()[] for more details.
 
-[NOTE]
-====
-There're some known limitations on JUnit 5, e.g. tests in static nested classes won't be discovered and classes are still displayed by its name instead of `@DisplayName`.
-They'll be fixed in future version of Gradle. If you find more, please don't hesitate to tell us: https://github.com/gradle/gradle/issues/new
-====
```
Comment (JLLeitschuh) [inline subprojects/docs/src/docs/userguide/javaPlugin.adoc]: I'm assuming that this was fixed in some other work but this was never removed from the documentation?

**Comment by JLLeitschuh** [comment]
Might not in the scope of this PR, but do you want to add a mention of filtering tags? Either in this PR or in a follow-on?

**Comment by blindpirate** [comment]
@JLLeitschuh Well, I believe `includeTags`/`excludeTags` are already in the documentation - https://docs.gradle.org/4.6-rc-2/userguide/java_plugin.html#test_grouping Thanks for your reminding.

**Comment by eriwen** [comment]
Looks good @blindpirate. Thank you

File: subprojects/docs/src/docs/userguide/javaPlugin.adoc [hunk 2]
```diff
@@ -799,6 +793,20 @@ In this way, you can use `gradle test` to test JUnit 3/4 tests on JUnit Platform
 
 A sample of mixed tests can be found at `samples/testing/junitplatform/engine` in the '-all' distribution of Gradle.
 
+[[custom_test_engine]]
+===== Custom test engine
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/javaPlugin.adoc]: engines

File: subprojects/docs/src/docs/userguide/javaPlugin.adoc [hunk 3]
```diff
@@ -799,6 +793,20 @@ In this way, you can use `gradle test` to test JUnit 3/4 tests on JUnit Platform
...
+
+With JUnit Platform, you can customize `TestEngine` implementation yourself. By default, `TestEngine` registration is done via Java's `java.util.ServiceLoader` mechanism.
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/javaPlugin.adoc]: JUnit platform allows you to use different test engines.

File: subprojects/docs/src/docs/userguide/javaPlugin.adoc [hunk 4]
```diff
@@ -799,6 +793,20 @@ In this way, you can use `gradle test` to test JUnit 3/4 tests on JUnit Platform
...
+In most cases, all you need to do is registering your engine implementation as documented https://junit.org/junit5/docs/current/user-guide/#launcher-api-engines-custom[here].
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/javaPlugin.adoc]: Not in most cases, it is always done via a ServiceLoader. The include/exclude filter simply allows you to filter the set of engines that will be used. Instead we could shorten this to:

By default, all engines on the test runtime classpath will be used. If you want to control the engines used explicitly, you can do as follows:

File: subprojects/docs/src/samples/testing/junitplatform/engine/build.gradle [hunk 5]
```diff
@@ -13,9 +13,11 @@ dependencies {
 }
 // END SNIPPET vintage-dependencies
 
+// START SNIPPET filter-engine
 test {
     useJUnitPlatform {
         // includeEngines 'included-engine'
```
Comment (oehme) [inline subprojects/docs/src/samples/testing/junitplatform/engine/build.gradle]: It's somewhat confusing that the example uses commented out includes/excludes.

How about having a real engine filtering sample where we use at least one of them without a comment? 

Also unrelated to this commit, but the snippet above is misleading - If you just want vintage, you don't need the jupiter api and engine on the classpath. Seems like we are trying to put too many things into one example.

File: subprojects/docs/src/docs/userguide/javaPlugin.adoc (hunk 2 above)
Comment (blindpirate) [reply subprojects/docs/src/docs/userguide/javaPlugin.adoc]: Done.

File: subprojects/docs/src/docs/userguide/javaPlugin.adoc (hunk 3 above)
Comment (blindpirate) [reply subprojects/docs/src/docs/userguide/javaPlugin.adoc]: Done.

File: subprojects/docs/src/samples/testing/junitplatform/engine/build.gradle (hunk 5 above)
Comment (blindpirate) [reply subprojects/docs/src/samples/testing/junitplatform/engine/build.gradle]: Yes. Previously the mixed JUnit3/4/5 sample and engine sample are put together. Now I've separated them as different samples - 'mix' and 'engine'.

Comment (blindpirate) [reply subprojects/docs/src/samples/testing/junitplatform/engine/build.gradle]: Yes. Previously the mixed JUnit3/4/5 sample and engine sample are put together. Now I've separated them as different samples - 'mix' and 'engine'.
